### PR TITLE
TE : adding email migration script

### PR DESCRIPTION
**Pull request by puneetjaiswal**

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/tools/Hibernate2JDBCMigration.java [hunk 1]
```diff
@@ -27,6 +29,32 @@ public void run() throws Exception {
...
+
+  private void cloneEmailConfigs() {
+    EmailConfigurationManager hibernateManager = PersistenceUtil
+        .getInstance(com.linkedin.thirdeye.datalayer.bao.hibernate.EmailConfigurationManagerImpl.class);
+    EmailConfigurationManager jdbcManager = DaoProviderUtil
+        .getInstance(com.linkedin.thirdeye.datalayer.bao.jdbc.EmailConfigurationManagerImpl.class);
+    List<EmailConfigurationDTO> dtoSrcList = hibernateManager.findAll();
+    List<EmailConfigurationDTO> dtoTgtList = jdbcManager.findAll();
+    Set<String> uniques = new HashSet<>();
+    //ALTER TABLE `webapp_config_index` ADD UNIQUE `unique_index`(`name`, `collection`, `type`);
```
Comment (kishoreg) [inline thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/tools/Hibernate2JDBCMigration.java]: remove this comment or change to reflect the unique constraint for email_config

File: thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/tools/Hibernate2JDBCMigration.java [hunk 2]
```diff
@@ -27,6 +29,32 @@ public void run() throws Exception {
...
+    for (EmailConfigurationDTO dto : dtoTgtList) {
+      String key = dto.getCollection() + "_" + dto.getMetric();
+      uniques.add(key);
+    }
+    for (EmailConfigurationDTO dto : dtoSrcList) {
+      String key = dto.getCollection() + "_" + dto.getMetric();
+      if (!uniques.contains(key)) {
+        dto.setId(null);
+        jdbcManager.save(dto);
```
Comment (kishoreg) [inline thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/tools/Hibernate2JDBCMigration.java]: email config references the function ids as well. We need to compute the mapping with new_function_id and set emailConfigurationDTO.setFunctionIds before calling save

File: thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/tools/Hibernate2JDBCMigration.java [hunk 3]
```diff
@@ -27,6 +29,32 @@ public void run() throws Exception {
...
+      } else {
+        System.out.println(
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/tools/Hibernate2JDBCMigration.java]: Do we need to use logger here? Or it's fine to print on screen.

File: thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/tools/Hibernate2JDBCMigration.java (hunk 2 above)
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/tools/Hibernate2JDBCMigration.java]: added email/function mapping through UI.

File: thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/tools/Hibernate2JDBCMigration.java (hunk 3 above)
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/tools/Hibernate2JDBCMigration.java]: yes, updated in another PR.

**Pull request merged**